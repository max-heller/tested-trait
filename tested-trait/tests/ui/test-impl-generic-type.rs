use tested_trait::{test_impl, tested_trait};

fn main() {
    #[tested_trait]
    trait Foo {}

    #[test_impl]
    #[in_integration_test]
    impl<T> Foo for Vec<T> {}
}
