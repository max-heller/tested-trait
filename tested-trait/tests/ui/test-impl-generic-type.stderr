error: #[test_impl] on a generic impl requires specifying concrete implementations with #[test_impl(Vec < T >: Foo)]
 --> tests/ui/test-impl-generic-type.rs:7:5
  |
7 |     #[test_impl]
  |     ^^^^^^^^^^^^
  |
  = note: this error originates in the attribute macro `test_impl` (in Nightly builds, run with -Z macro-backtrace for more info)

error: associated tests for this generic implementation can only be instantiated for concrete types
 --> tests/ui/test-impl-generic-type.rs:9:9
  |
9 |     impl<T> Foo for Vec<T> {}
  |         ^^^
